ir/state.h

```diff
@@ -107,6 +108,16 @@ class State {
   };
   std::map<std::string, std::map<FnCallInput, FnCallOutput>> fn_call_data;
 
+  // src -> <(dst, cond), isolated ub>
+  std::unordered_map<const BasicBlock*, std::pair<std::vector<std::pair<
+                                                              const BasicBlock*,
+                                                              smt::expr>>,
+                                                              std::optional<
+                                                              smt::expr>>>
```

Comment:
What about declaring a separate struct (just above this field definition) that stores the data instead of using std::pair< ... >?

Reply:
Thanks!, i changed it and it should be a bit more readable now.